url.go

```diff
@@ -41,17 +47,19 @@ type URL struct {
 	Series   string // "precise" or "" if unset; "bundle" if it's a bundle.
 }
 
+const bundleSeries = "bundle"
+
 var ErrUnresolvedUrl error = fmt.Errorf("charm or bundle url series is not resolved")
 
 var (
-	validSeries = regexp.MustCompile("^[a-z]+([a-z0-9]+)?$")
+	validSeries = set.NewStrings(series.SupportedSeries()...).Union(set.NewStrings(bundleSeries))
```

Comment:
This is probably ok but introduces the risk of incorrect behaviour when a new Ubuntu series is created. Every time this happens (every 6 months) thing might not work right until Juju is updated and released, or /usr/share/distro-info/ubuntu.csv is updated.

That said, I can't think of a better solution given the constraints, except perhaps having the parsing function return all possible URLs given the input and then querying the charmstore to determine which ones actually exist. That sounds pretty awful though.